tests/common/util.rs

```diff
@@ -1398,15 +1477,27 @@ impl UCommand {
                 stdio
             };
 
-            self.raw
+            command
                 .stdin(self.stdin.take().unwrap_or_else(Stdio::null))
                 .stdout(stdout)
-                .stderr(stderr)
+                .stderr(stderr);
         };
 
+        (command, captured_stdout, captured_stderr)
+    }
+
+    /// Spawns the command, feeds the stdin if any, and returns the
+    /// child process immediately.
+    pub fn run_no_wait(&mut self) -> UChild {
+        assert!(!self.has_run, "{}", ALREADY_RUN);
```

Comment:
Since this is a wrapper now that builds a Command on every run, we don't need to check this anymore do we? That's kind of a cool side effect of doing it this way.

Reply:
Actually, we do need to check this because currently `build` mutates `self`, but it doesn't have to! We can write it in a way that it doesn't do that and then we can make reusable commands, which could be nice in some cases where we test many small variations on one command.

Reply:
I have thought about it, too. There's a problem however: `Stdio` is not cloneable, so every `Stdio` field we would set explicitely with `set_stdin`, `set_stdout` and `set_stderr` would be gone after the first usage. I'm not sure if this behavior is obvious enough. Another possibility to achieve almost the same would be to add a method like `from_other(other: &UCommand, stdin: Option<Stdio>, stdout: ...)`.

Reply:
Oh that's a good point! Nevermind then :)

Reply:
Maybe the opposite approach then, could we make `build` consume `self`?